backend.go

```diff
@@ -15,12 +18,42 @@ func NewFilesystemBackend(root, tmp string) (storage.Backend, error) {
 		return nil, err
 	}
 
+	storage, err := findAllBackends(fsobj, packs, root)
+	if err != nil {
+		return nil, err
+	}
+
 	return &filesystemBackend{
 		fs:       fsobj,
-		backends: []storage.Storage{fsobj, packs},
+		backends: storage,
 	}, nil
 }
 
+func findAllBackends(mainLoose *fileStorer, mainPacked *pack.Storage, root string) ([]storage.Storage, error) {
+	storage := make([]storage.Storage, 2)
+	storage[0] = mainLoose
+	storage[1] = mainPacked
+	f, err := os.Open(path.Join(root, "info", "alternates"))
+	if err != nil {
+		// No alternates file, no problem.
+		if err != os.ErrNotExist {
+			return storage, nil
+		}
+		return nil, err
+	}
+
+	scanner := bufio.NewScanner(f)
+	for scanner.Scan() {
+		storage = append(storage, newFileStorer(scanner.Text(), ""))
+		pack, err := pack.NewStorage(scanner.Text())
+		if err != nil {
+			return nil, err
+		}
+		storage = append(storage, pack)
+	}
+	return storage, nil
```

Comment:
It's possible that we will have to handle an error here, too. Here's an example:

```go
scanner := bufio.NewScanner(...)
for scanner.Scan() {
        // ...
}

if err := scanner.Err(); err != nil {
        return nil, err
}
```

Reply:
Good point.